frontend/src/pages/ExploreData/ExploreDataPage.tsx

```diff
@@ -216,15 +215,21 @@ function ExploreDataPage(props: ExploreDataPageProps) {
 
   // Set up sticky madlib behavior
   const [isSticking, setIsSticking] = useState<boolean>(false)
-  useScrollPosition(
-    ({ pageYOffset, stickyBarOffsetFromTop }) => {
-      const topOfMadLibContainer = pageYOffset > stickyBarOffsetFromTop
-      if (topOfMadLibContainer && !noTopicChosen) setIsSticking(true)
-      else setIsSticking(false)
-    },
-    [isSticking],
-    300
-  )
+  const madlibRef = useCallback((node: HTMLDivElement | null) => {
+    if (node) {
+      // https://stackoverflow.com/a/61115077/7902371
+      const observer = new IntersectionObserver(
+        ([e]) => setIsSticking(e.intersectionRatio < 1),
+        {
+          rootMargin: '-1px 0px 0px 0px',
+          threshold: [1],
+        }
+      );
+
+      observer.observe(node);
+    }
+  }, [])
```

Comment:
I've found that the `useScrollPosition` isn't very accurate. It only triggers the sticky behaviour when the top exceeds by a few pixels.